diff --git a/controllers/argocd/util.go b/controllers/argocd/util.go
index d9f0859..01fffc8 100644
--- a/controllers/argocd/util.go
+++ b/controllers/argocd/util.go
@@ -646,7 +646,12 @@ func nameWithSuffix(suffix string, cr *argoproj.ArgoCD) string {
 // fqdnServiceRef will return the FQDN referencing a specific service name, as set up by the operator, with the
 // given port.
 func fqdnServiceRef(service string, port int, cr *argoproj.ArgoCD) string {
-       return fmt.Sprintf("%s.%s.svc.cluster.local:%d", nameWithSuffix(service, cr), cr.Namespace, port)
+       // FIXME: This is possibly anti-pattern
+       dnsSuffix, exists := os.LookupEnv("CLUSTER_DNS_DOMAIN")
+       if !exists || strings.TrimSpace(dnsSuffix) == "" {
+               dnsSuffix = "cluster.local"
+       }
+       return fmt.Sprintf("%s.%s.svc.%s:%d", nameWithSuffix(service, cr), cr.Namespace, dnsSuffix, port)
 }

 // InspectCluster will verify the availability of extra features available to the cluster, such as Prometheus and
